src/main/java/potong/TaskList.java

```diff
@@ -0,0 +1,147 @@
+        if (index - 1 >= this.arr.size()) {
+            throw new PotongException("We cannot unmark a task thats not there!");
+        }
+        return this.arr.get(index - 1).unmark();
+    }
+
+    /**
+     * Delete a task in the list.
+     *
+     * @param index Index of the task.
+     * @return String representation of the action.
+     * @throws PotongException If the index is out of bounds.
+     */
+    public String delete(int index) throws PotongException {
+        if (index - 1 >= this.arr.size()) {
+            throw new PotongException("We cannot delete a task thats not there!");
+        }
+        Task removed = this.arr.remove(index - 1);
+        return String.format("Noted. I've removed this task:\n %s\nNow you have %d tasks in the list.",
+                removed, this.arr.size());
+    }
+
+    /**
+     * Save the list of tasks.
+     *
+     * @param storage Storage class to save the list.
+     * @throws IOException If the input is wrong.
+     */
+    public void writeToStorage(Storage storage) throws IOException {
+        StringBuilder result = new StringBuilder();
+        for (Task curr : this.arr) {
+            String type = curr.getType();
+            String status = curr.getStatus();
+            String description = curr.getDescription();
+            String time = curr.getTime();
+            if (time.isEmpty()) {
+                String task = String.format("%s | %s | %s\n", type, status, description);
+                result.append(task);
+            } else {
+                String task = String.format("%s | %s | %s | %s\n", type, status, description, time);
+                result.append(task);
+            }
+        }
+        storage.writeToFile(result.toString());
+    }
+
+    /**
+     * Find all tasks with the keyword in the list of tasks.
+     *
+     * @param keyword Keyword.
+     * @return String representation of the list of tasks with the keyword.
+     */
+    public String find(String keyword) {
+        ArrayList<Task> result = new ArrayList<>();
+        for (Task curr : this.arr) {
+            if (curr.findKeyword(keyword)) {
+                result.add(curr);
+            }
+        }
+        StringBuilder stringResult = new StringBuilder("Here are the matching tasks in your list:\n");
+        for (int i = 0; i < result.size(); i++) {
+            stringResult.append(String.format("%d. %s\n", i + 1, result.get(i)));
+        }
+        return stringResult.toString();
+    }
+
+    /**
+     * String representation of the list of tasks.
+     *
+     * @return String representation of this list.
+     */
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 0; i < this.arr.size(); i++) {
+            result.append(String.format("%d. %s\n", i + 1, this.arr.get(i)));
+        }
+        return result.toString();
+    }
+}
```

Comment:
I like how you have added JavaDocs comments for all your methods, it makes it very easy to understand your code.